onelogin_aws_cli/__init__.py

```diff
@@ -178,11 +178,15 @@ def save_credentials(self):
         if "profile" in self.config:
             name = self.config["profile"]
 
-        cred_config[name] = {
-            "aws_access_key_id": creds["AccessKeyId"],
-            "aws_secret_access_key": creds["SecretAccessKey"],
-            "aws_session_token": creds["SessionToken"]
-        }
+        # Set each value specifically instead of overwriting the entire
+        # profile block in case they have other parameters defined
+        cred_config[name]['aws_access_key_id'] = creds["AccessKeyId"]
```

Comment:
Won't this fail if the user has no entry in the config with this profile name?   Adding the following would probably fix it:

```
cred_config[name] = {}
```

Reply:
I added conditional code to check if the key exists in the dictionary and initialize if it does not.